### ISSUE TITLE
Question regarding firmware

### ISSUE THREAD

**Issue by skaankhunt** [clmnsr/Klipper-for-Sovol-SV01-Pro#3]
Well done!
I'm sat here trying to do my first Klipper installation on my bone stock SV01 Pro. 

I've currently installed Mainsail OS from the Raspberry Pi OS imager, and flashed it onto my Raspberry Pi 4B. I've then uploaded all the files you listed onto my Mainsail page.

Now I'm following a Youtube guide, in which he uses ssh to open the klipper directory and runs "make menuconfig" where he does some configurations regarding his microcontroller aswell as the printers mainboard.

To my question.. Is this already configured now that I've copied your config files, or do I need to do this step as well? I will admit it looks a bit tricky..

Also if I'm missing anything else I'd very much appreciate a nod in the right direction.

**Comment by clmnsr** [comment]
Hi @skaankhunt,

thank you for using my klipper-Config!
I´ve already compiled the firmware for you. This is the compiled firmware: [klipper-010292.bin](https://github.com/clmnsr/Klipper-for-Sovol-SV01-Pro/blob/main/klipper-010292.bin)

Put this file on a SDcard and flash the printer. Use my printer.cfg and you´re done!

**Comment by skaankhunt** [comment]
Nice, thank you! I think I got a bit lost regarding the firmware spoken about in the video. Just to clarify.. 

I was under the impression that the .bin fin file you uploaded we're just some firmware for the printer allowing for running klipper, and I though that all the configs you've posted were files I should upload to my Raspberry Pi.

But from you're message I get that all of the configs (apart from printer.cfg?) are compiled and installed onto the printer when flashing the software with the SD-card, but I think those configs can be accessed through the Pi after the install. 

Am I on the right track? Thank you for your patience :D

**Comment by clmnsr** [comment]
No, no.

First step is to flash the firmware.bin.
Second step to install klipper, mainsail, moonraker.
Third step to put the configs (without the firmware.bin) into the printer_data/configs folder.

After that you can reload the firmware in Mainsail OS and the printer is ready!

**Comment by skaankhunt** [comment]
Oh haha I see. I think I got it sorted but now I have a warning in mainsail, saying 
"Include file '/home/pi/SV01PRO_data/config/macros.cfg' does not exist."

When I SSH I can find a directory /printer_data/config which seems to include all your files.

I assume I will need to change the include wherever it is? In the printer.cfg?

Also another question.. Should I replace the mainsail.cfg i find in my Mainsail dashboard under "Machine" with yours or leave it stock?

**Comment by clmnsr** [comment]
Yes, you need to change in printer.cfg the include pathes, if you have different pathes.
You have maybe diffent mainsail.cfg, just leave the file and don´t swap it. Important is only macro and printer.cfg.

**Comment by skaankhunt** [comment]
I can't even find an Include macros, all I've got in the printer.cfg is:
[include mainsail.cfg]
[include base.cfg]

**Comment by clmnsr** [comment]
Ah, I forgot it!

Put this into printer.cfg:
[include macros.cfg]

**Comment by skaankhunt** [comment]
Nice, I’m not at home right now so can’t test it sadly. Is that all I have to do or do I need to point on a file path somehow?

**Comment by clmnsr** [comment]
When the files (include macros.cfg) are in same directory than the printer.cfg you need only to add this line.

**Comment by skaankhunt** [comment]
Thank you again, I will try when i get home. Sadly I think it will complain about file paths considering the error i got. Anyways I will update you once I get home :D

**Comment by skaankhunt** [comment]
Update:
I appended [include macros.cfg] to the printer.cfg file, but the issue persisted complaining about the file path.
I then when into base.cfg and managed to edit the paths. In my case I had to just replace "SV01_PRO_data" with "printer_data".

Now it all seems to be working!
Thank your contributions and all the support you've given me :D
